File: Examples/VisualBasic/Articles/RenderingAndPrinting/SpecifyJobWhilePrinting.vb

Imports System.Collections.Generic
Imports System.Linq
Imports System.Text
Imports Aspose.Cells.Rendering

Namespace Articles.RenderingAndPrinting
    Public Class SpecifyJobWhilePrinting
        Public Shared Sub Run()
            'ExStart:1

            ' The path to the documents directory.
            Dim dataDir As String = RunExamples.GetDataDir(System.Reflection.MethodBase.GetCurrentMethod().DeclaringType)

            'Create workbook object from source Excel file
            Dim workbook As New Workbook(dataDir & Convert.ToString("SampleBook.xlsx"))

            'Specify Printer and Job Name
            Dim printerName As String = ""

            While String.IsNullOrEmpty(printerName) AndAlso String.IsNullOrWhiteSpace(printerName)
                Console.WriteLine("Please Enter Your Printer Name:")
                printerName = Console.ReadLine()
            End While

            Dim jobName As String = "Job Name while Printing with Aspose.Cells"

            'Print workbook using WorkbookRender
            Dim wr As New WorkbookRender(workbook, New ImageOrPrintOptions())
            Try
                wr.ToPrinter(printerName, jobName)
            Catch ex As Exception
                Console.WriteLine(ex.Message)
            End Try

            'Access first worksheet
            Dim worksheet As Worksheet = workbook.Worksheets(0)

            'Print worksheet using SheetRender
            Dim sr As New SheetRender(worksheet, New ImageOrPrintOptions())
            Try
                sr.ToPrinter(printerName, jobName)
            Catch ex As Exception
                Console.WriteLine(ex.Message)
            End Try

            'ExEnd:1
        End Sub
    End Class
End Namespace

File: Examples/CSharp/Articles/RenderingAndPrinting/SpecifyJobWhilePrinting.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using Aspose.Cells.Rendering;

namespace Aspose.Cells.Examples.CSharp.Articles.RenderingAndPrinting
{
    public class SpecifyJobWhilePrinting
    {
        public static void Run()
        {
            Console.WriteLine("SpecifyJobWhilePrinting Started Now.");

            //Source directory
            string sourceDir = RunExamples.Get_SourceDirectory();

            // Create workbook object from source Excel file
            Workbook workbook = new Workbook(sourceDir + "sampleSpecifyJobWhilePrinting.xlsx");

            string printerName = "doPDF 8";

            string jobName = "My Job Name";

            // Print workbook using WorkbookRender
            WorkbookRender wr = new WorkbookRender(workbook, new ImageOrPrintOptions());
            try
            {
                wr.ToPrinter(printerName, jobName);
            }
            catch (Exception ex)
            {
                Console.WriteLine(ex.Message);
            }

            Console.Write("Press Enter to continue...");
            Console.ReadLine();

            // Access first worksheet
            Worksheet worksheet = workbook.Worksheets[1];

            // Print worksheet using SheetRender
            SheetRender sr = new SheetRender(worksheet, new ImageOrPrintOptions());
            try
            {
                sr.ToPrinter(printerName, jobName);
            }
            catch (Exception ex)
            {
                Console.WriteLine(ex.Message);
            }

            Console.WriteLine("SpecifyJobWhilePrinting executed successfully.");
        }
    }
}
